PixArt/diffusers_convert.py

```diff
@@ -40,24 +16,53 @@
 	("ar_embedder.mlp.2.bias",   "adaln_single.emb.aspect_ratio_embedder.linear_2.bias"),
 ]
 
-# Add actual transformer blocks
-for depth in range(28):
-	# Transformer blocks
-	conversion_map += [
-		(f"blocks.{depth}.scale_shift_table", f"transformer_blocks.{depth}.scale_shift_table"),
-		# Projection
-		(f"blocks.{depth}.attn.proj.weight", f"transformer_blocks.{depth}.attn1.to_out.0.weight"),
-		(f"blocks.{depth}.attn.proj.bias",   f"transformer_blocks.{depth}.attn1.to_out.0.bias"),
-		# Feed-forward
-		(f"blocks.{depth}.mlp.fc1.weight", f"transformer_blocks.{depth}.ff.net.0.proj.weight"),
-		(f"blocks.{depth}.mlp.fc1.bias",   f"transformer_blocks.{depth}.ff.net.0.proj.bias"),
-		(f"blocks.{depth}.mlp.fc2.weight", f"transformer_blocks.{depth}.ff.net.2.weight"),
-		(f"blocks.{depth}.mlp.fc2.bias",   f"transformer_blocks.{depth}.ff.net.2.bias"),
-		# Cross-attention (proj)
-		(f"blocks.{depth}.cross_attn.proj.weight" ,f"transformer_blocks.{depth}.attn2.to_out.0.weight"),
-		(f"blocks.{depth}.cross_attn.proj.bias"   ,f"transformer_blocks.{depth}.attn2.to_out.0.bias"),
+def get_depth(state_dict):
+	return sum(key.endswith('.scale_shift_table') for key in state_dict.keys())

```

Comment:
See the comment for `loader.py`. There is one extra `scale_shift_table` entry, meaning it'll be +1. I think `.attn1.to_out.0.weight` is a good key to count.